File: advanced-programming-methods/toy-interpreter-java/src/model/statement/PrintStatement.java
package model.statement;

import model.expression.Expression;

/**
 * The type Print statement.
 */
public class PrintStatement extends Statement
{
    private Expression expression;

    /**
     * Instantiates a new Print statement.
     *
     * @param expression the expression
     */
    public PrintStatement(Expression expression)
    {
        this.expression = expression;
    }

    /**
     * Getter for property 'expression'.
     *
     * @return Value for property 'expression'.
     */
    public Expression getExpression()
    {
        return expression;
    }

    /** {@inheritDoc} */
    @Override
    public String toString()
    {
        return String.format("Print(%s)", expression.toString());
    }

    /** {@inheritDoc} */
    @Override
    public Statement cloneDeep()
    {
        return new PrintStatement(expression.cloneDeep());
    }
}

File: advanced-programming-methods/labs/toy-interpreter-sharp/model/statement/PrintStatement.cs
using System;

namespace Model
{
    /// <summary>
    /// The type Print statement.
    /// </summary>
    [Serializable()]
    public class PrintStatement : Statement
    {
        private readonly Expression expression;

        /// <summary>
        /// Instantiates a new Print statement.
        /// </summary>
        /// <param name="expression"> the expression </param>
        public PrintStatement(Expression expression)
        {
            this.expression = expression;
        }

        /// <summary>
        /// Getter for property 'expression'.
        /// </summary>
        /// <returns> Value for property 'expression'. </returns>
        public virtual Expression Expression
        {
            get
            {
                return expression;
            }
        }

        /// <inheritdoc />
        public override string ToString()
        {
            return string.Format("Print({0})", expression);
        }

        /// <inheritdoc />
        public override Statement CloneDeep()
        {
            return new PrintStatement(expression.CloneDeep());
        }
    }

}
